src/base/logger.cpp

```diff
@@ -124,6 +156,45 @@ QVector<Log::Peer> Logger::getPeers(const int lastKnownId) const
     return loadFromBuffer(m_peers, (size - diff));
 }
 
+std::pair<int, int> Logger::getMessagesRange(const int lastKnownId) const
```

Comment:
IMO it is really hard to catch up with its logic, maybe it is better to have 2 functions:
1. `std::pair<int, int> Logger::getMessagesRange() const`: that simply return {start, end} ID ranges of `m_messages`.
2. Helper function `std::pair<int, int> getValidMessagesRange(const int lastKnownId) const` this could be at logfiltermodel.cpp (outside of logger.cpp), that computes valid ranges from `Logger::getMessagesRange()`

Same applies to the Peers part.

Reply:
This returns the valid messages from a known last ID. (or none ID if you want all the range).
It is basically the counterpart of `getMessages()`